<?xml version="1.0"?>
<launch>
  <arg name="uri1" default="radio://0/120/2M/E7E7E7E707" /> <!--Adjust this parameter-->
  <arg name="frame1" default="cf1" /> <!--For OptiTrack users: frame name should match the Rigid-body name on Motive-->
  <arg name="ip" default="169.254.148.86" /> <!--Adjust this paramter-->
  <arg name="port" default="3883" />

  <include file="$(find crazyflie_driver)/launch/crazyflie_server.launch">
  </include>

  <group ns="cf1">
    <node pkg="crazyflie_driver" type="crazyflie_add" name="crazyflie_add" output="screen">
      <param name="uri" value="$(arg uri1)" />
      <param name="tf_prefix" value="$(arg frame1)" />
      <param name="enable_logging" value="False" /> <!--Logging is disabled to reduce unnecessary bandwith consumption-->
      <param name="enable_logging_imu" value="False" />
      <param name="enable_logging_temperature" value="False" />
      <param name="enable_logging_magnetic_field" value="False" />
      <param name="enable_logging_pressure" value="False" />
      <param name="enable_logging_battery" value="False" />
      <param name="enable_logging_packets" value="False" />
      <rosparam>
        genericLogTopics: ["log1"]
        genericLogTopicFrequencies: [100]
        genericLogTopic_log1_Variables: ["stateEstimate.x", "stateEstimate.y", "stateEstimate.z"]
      </rosparam>
    </node>

    <!--
    <node name="pose" pkg="crazyflie_demo" type="publish_external_position_vrpn.py" output="screen">
      <param name="topic" value="/vrpn_client_node/cf1/pose" />
    </node>
    -->

    <include file="$(find crazyflie_controller)/launch/crazyflie2.launch">
    <arg name="frame" value="$(arg frame1)"/>
    </include>


    <!-- launch pose_publisher with given parameter -->
    <node name="dynamic_route" pkg="crazyflie_demo" type="dynamic_route.py" output="screen">
    <param name="tf_prefix" value="$(arg frame1)"/>
    <param name="frame" value="$(arg frame1)"/>
    <param name="name" value="goal"/>
    <param name="route" value="$(arg route)"/>
    <param name="rate" value="30"/>
    <!--
    <param name="x" value="$(arg x1)"/>
    <param name="y" value="$(arg y1)"/>
    <param name="z" value="$(arg z1)"/>
    -->
</node>

  </group>


  <!-- run a single vrpn client -->
  <node pkg="vrpn_client_ros" type="vrpn_client_node" name="vrpn_client_node" output="screen">
    <rosparam subst_value="true">
      server: $(arg ip)
      port: $(arg port)
      update_frequency: 100.0
      frame_id: world
      use_server_time: false
      broadcast_tf: true
      refresh_tracker_frequency: 1.0 <!-- VRPN will add all trackers found on connection -->
    </rosparam>
  </node>

 </launch>
